src/Ractive/prototype/set.js

```diff
@@ -10,9 +10,9 @@ export default function Ractive$set ( keypath, value ) {
 	if ( isObject( keypath ) ) {
 		const map = keypath;
 
-		for ( keypath in map ) {
-			if ( map.hasOwnProperty( keypath) ) {
-				set( this, keypath, map[ keypath ] );
+		for ( const k in map ) {
+			if ( map.hasOwnProperty( k) ) {
+				set( this, k, map[k] );
 			}
```

Comment:
one could find out pretty easily to kinda get an idea how much of a difference actually by first redefining the `Array.prototype.forEach`

just a minute, I'm gonna try it :)